packages/grid_client/src/zos/zmachine_light.ts

```diff
@@ -0,0 +1,84 @@
+import { Expose, Transform, Type } from "class-transformer";
+import {
+  IsBoolean,
+  IsDefined,
+  IsInt,
+  IsIP,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  IsUrl,
+  Max,
+  Min,
+  ValidateNested,
+} from "class-validator";
+
+import { ValidateMembers } from "../helpers";
+import { ComputeCapacity } from "./computecapacity";
+import { WorkloadData, WorkloadDataResult } from "./workload_base";
+import { Mount, MyceliumIP } from "./zmachine";
+
+class MachineInterface {
+  @Expose() @IsString() @IsNotEmpty() network: string;
+  @Expose() @IsIP() @IsNotEmpty() ip: string;
+}
+
+class ZmachineLightNetwork {
+  //what is ValidateNested
+  @Expose() @Type(() => MachineInterface) @ValidateNested({ each: true }) interfaces: MachineInterface[];
+  @Expose() @Type(() => MyceliumIP) @ValidateNested() mycelium: MyceliumIP;
+
+  challenge(): string {
+    let out = "";
+    for (let i = 0; i < this.interfaces.length; i++) {
+      out += this.interfaces[i].network;
+      out += this.interfaces[i].ip;
```

Comment:
```suggestion
      out += this.interfaces[i].ip || "";
```

Reply:
I will dismiss this as it's just coding style and I'm following the other zos classes as it might affect signatures and challenges